app/src/ereferrals/java/org/eyeseetea/malariacare/strategies/LoginActivityStrategy.java

```diff
@@ -111,6 +123,78 @@ public void onGetUsername(Credentials credentials) {
         });
     }
 
+    private void showSurveyWhenSoftLoginAndValidCredentials() {
+        if(loginActivity.getIntent().getStringExtra(SplashScreenActivity.INTENT_JSON_EXTRA_KEY)==null
+                || loginActivity.getIntent().getStringExtra(SplashScreenActivity.INTENT_JSON_EXTRA_KEY).isEmpty()) {
+            return;
+        }
+        IMainExecutor mainExecutor = new UIThreadExecutor();
+        IAsyncExecutor asyncExecutor = new AsyncExecutor();
+        ICredentialsRepository credentialsRepository = new CredentialsLocalDataSource();
+        GetLastInsertedCredentialsUseCase getLastInsertedCredentialsUseCase =
+                new GetLastInsertedCredentialsUseCase(mainExecutor, asyncExecutor,
+                        credentialsRepository);
+        getLastInsertedCredentialsUseCase.execute(new GetLastInsertedCredentialsUseCase.Callback() {
+            @Override
+            public void onGetUsername(Credentials credentials) {
+                if (credentials != null && !credentials.isEmpty()) {
```

Comment:
please, can you refactor this anonymous method for avoiding nested ifs (for example creating functions) for has a more readable code?